src/ngraph/state/rng_state.cpp

```diff
@@ -0,0 +1,44 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <random>
+
+#include "except.hpp"
+#include "rng_state.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+void ngraph::RNGState::activate()
+{
+    //reentrant?
+    if (!is_active())
```

Comment:
Either throw an error if it's already active, or keep a counter for how active it is so that deactivate doesn't happen to early. Probably throwing an error is correct for now.